lib/src/layout_element.dart

```diff
@@ -32,27 +32,35 @@ class TableLayoutElement extends LayoutElement {
   @override
   Widget toWidget(RenderContext context) {
     final rows = <TableRowLayoutElement>[];
-    List<TrackSize> columnSizes;
+    List<TrackSize> columnSizes = <TrackSize>[];
     for (var child in children) {
       if (child is TableStyleElement) {
         // Map <col> tags to predetermined column track sizes
-        columnSizes = child.children.where((c) => c.name == "col").map((c) {
-          final colWidth = c.attributes["width"];
-          if (colWidth != null && colWidth.endsWith("%")) {
-            final percentageSize =
-                double.tryParse(colWidth.substring(0, colWidth.length - 1));
-            return percentageSize != null
-                ? FlexibleTrackSize(percentageSize * 0.01)
-                : FlexibleTrackSize(1);
-          } else if (colWidth != null) {
-            final fixedPxSize = double.tryParse(colWidth);
-            return fixedPxSize != null
-                ? FixedTrackSize(fixedPxSize)
-                : FlexibleTrackSize(1);
-          } else {
-            return FlexibleTrackSize(1);
-          }
-        }).toList(growable: false);
+        columnSizes = child.children
+            .where((c) => c.name == "col")
+            .map((c) {
+              final span =
+                  int.parse(c.attributes["span"] ?? "1", onError: (_) => 1);
+              final colWidth = c.attributes["width"];
+              return List.generate(span, (index) {
+                if (colWidth != null && colWidth.endsWith("%")) {
+                  final percentageSize = double.tryParse(
+                      colWidth.substring(0, colWidth.length - 1));
+                  return percentageSize != null
+                      ? FlexibleTrackSize(percentageSize * 0.01)
+                      : FlexibleTrackSize(1);
+                } else if (colWidth != null) {
+                  final fixedPxSize = double.tryParse(colWidth);
+                  return fixedPxSize != null
+                      ? FixedTrackSize(fixedPxSize)
+                      : FlexibleTrackSize(1);
+                } else {
+                  return FlexibleTrackSize(1);
+                }
+              });
+            })
+            .expand((element) => element)
```

Comment:
Is there any way that we could reduce the `.map` and `.expand` calls into a single `.expand` call? I feel like I almost understand what is going on here, but I want to double check.

Reply:
Ah, actually, is this so that colgroups work? If that's the case, LGTM

Reply:
Yes indeed, this is so that multiple colgroups work.